src/number_theory/bernoulli_polynomials.lean

```diff
@@ -122,6 +120,57 @@ begin
   rw [g, zero_smul],
 end
 
+/-- Another version of `polynomial.sum_bernoulli`. -/
+lemma bernoulli_eq_sub_sum (n : ℕ) : (n.succ : ℚ) • bernoulli n = monomial n (n.succ : ℚ) -
+  ∑ k in finset.range n, ((n + 1).choose k : ℚ) • bernoulli k :=
+begin
+  change _ = (monomial n) ((n : ℚ) + 1) - ∑ (k : ℕ) in range n,
+    ↑((n + 1).choose k) • bernoulli k,
+  rw [←sum_bernoulli n, sum_range_succ, add_comm],
+  simp only [cast_succ, choose_succ_self_right, add_sub_cancel],
+end
+
+/-- Another version of `bernoulli.sum_range_pow`. -/
+lemma sum_range_pow_eq_bernoulli_sub (n p : ℕ) :
+  (p  + 1 : ℚ) * ∑ k in range n, (k : ℚ) ^ p = (bernoulli p.succ).eval n -
+  (_root_.bernoulli p.succ) :=
+begin
+  rw [sum_range_pow, bernoulli_def, eval_finset_sum, ←sum_div, mul_div_cancel' _ _],
+  { simp_rw [eval_monomial],
+    symmetry,
+    rw [←sum_flip _, sum_range_succ],
+    simp only [tsub_self, tsub_zero, choose_zero_right, cast_one, mul_one, pow_zero,
+      add_tsub_cancel_right],
+    apply sum_congr rfl (λ x hx, _),
+    apply congr_arg2 _ (congr_arg2 _ _ _) rfl,
+    { rw nat.sub_sub_self (mem_range_le hx), },
+    { rw ←choose_symm (mem_range_le hx), }, },
+  { norm_cast, apply succ_ne_zero _, },
+end
+
+/-- Rearrangement of `polynomial.sum_range_pow_eq_bernoulli_sub`. -/
+lemma bernoulli_succ_eval (n p : ℕ) : (bernoulli p.succ).eval n =
+  _root_.bernoulli (p.succ) + (p  + 1 : ℚ) * ∑ k in range n, (k : ℚ) ^ p :=
+by { apply eq_add_of_sub_eq', rw sum_range_pow_eq_bernoulli_sub, }
+
+lemma bernoulli_eval_one_add (n : ℕ) (x : ℚ) :
+  (bernoulli n).eval (1 + x) = (bernoulli n).eval x + n * x^(n - 1) :=
+begin
+  apply nat.strong_induction_on n (λ d hd, _),
+  apply (mul_right_inj' _).1,
+  swap 3, exact d.succ,
```

Comment:
Can you pass `d.succ` directly as argument on the line above?

Reply:
Done!